### ISSUE TITLE
Prompt to Rebuild Container when devcontainer definition changes - from inside Container

### ISSUE THREAD

**Issue by daulis** [microsoft/vscode-remote-release#7752]
# Existing Related Feature
- From Host context
  - Start with some folder that has a .devcontainer definition
  - Open in Visual Studio Code
  - "Reopen in Container" - Context switches to the Dev Container
  - Close VS Code
  - Change devcontainer.json
  - "Reopen in Container"
- From Dev Container context
  - User prompted to "Rebuild"
- The existing feature works well. (It would be nice to combine the Reopen/Rebuild steps, but that is a separate feature request: https://github.com/microsoft/vscode-remote-release/issues/2235)

# Proposed Feature
- From Host context
  - Start with some folder that has a .devcontainer definition
  - Open in Visual Studio Code
  - "Reopen in Container" - Context switches to the Dev Container
- From Dev Container context
  - Change devcontainer.json
  - (NEW) User prompted to "Rebuild"
![image](https://user-images.githubusercontent.com/6913228/208704841-6bcefca0-9d76-48c5-b449-04d534340c4d.png)

# Why?
- Users will often work inside the Dev Container context and switch branches, git pull to latest on main branch, etc
- Users may not be aware that the devcontainer definition changed
- Features/building/etc may break because code in a project relies on the devcontainer definition

# Versions
- Visual Studio Code - 1.74.1
- Dev Containers v0.266.1
- OS: Windows 10 (but my Host is actually WSL)

**Comment by chrmarti** [comment]
Related: https://github.com/microsoft/vscode-remote-release/issues/7592

**Comment by daulis** [comment]
@chrmarti Awesome! I confirmed this works with Visual Studio Code 1.75.0, and Dev Containers
v0.275.0, so I'm closing this.

Side Note:
Also, awesome and thank you on the entire product. Visual Studio Code+DevContainers make SUCH a great workflow.

I actually rolled out a DevContainers workflow for my org of 100s of engineers this past week, transitioning from Windows-based compilation --> Linux containers. The overall developer experience is night and day. So, also thanks for making me look great at work :)
